tests/test_index_composite.py

```diff
@@ -16,6 +16,43 @@
 
 from common_faiss_tests import get_dataset_2
 
+
+class TestRemoveFastScan(unittest.TestCase):
+    def do_test(self, ntotal, removed):
+        d = 20
+        xt, xb, _ = get_dataset_2(d, ntotal, ntotal, 0)
+        index = faiss.index_factory(20, 'IDMap2,PQ5x4fs')
+        index.train(xt)
+        index.add_with_ids(xb, np.arange(ntotal).astype("int64"))
+        before = index.reconstruct_n(0, ntotal)
+        index.remove_ids(np.array(removed))
+        for i in range(ntotal):
+            if i in removed:
+                try:
+                    after = index.reconstruct(i)
```

Comment:
re-use of identifier i
these should all fail, correct?

Reply:
yes, if this vector i removed before, reconstruction should be failed.
I am not sure what do you mean by "re-use of identifier i"?